pkg/serviceaccount/legacy.go

```diff
@@ -126,6 +140,32 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 			klog.V(4).Infof("Service account UID no longer matches %s/%s: %q != %q", namespace, serviceAccountName, string(serviceAccount.UID), serviceAccountUID)
 			return nil, fmt.Errorf("ServiceAccount UID (%s) does not match claim (%s)", serviceAccount.UID, serviceAccountUID)
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LegacyServiceAccountTokenTracking) {
```

Comment:
this block also needs to be guarded on a check that the secret is referenced from the serviceAccount's secrets list (which indicates it actually is an auto-generated token)

make sure there's test coverage that manually created secrets don't get tracking annotations/labels or get warned about (I think the integration test is incorrect here)

Reply:
enable tracking for all legacy tokens are useful from security perspective. we make sure only auto-created tokens will be cleaned up once CleanUp feature is enabled.

Reply:
we don't need to protect it as security property. this adds observability.

agree to avoid emitting warning for manual created ones.

Reply:
done for warning bit.

Reply:
still need to ensure use of the manually created token in the integration test doesn't return a warning